---
title: "Periodontology 2000 - Script 2, Main Figures" 
author: "Diana Proctor"
date: "October 8, 2018"
output: html_document


---

### Import the data (from the decontamination script) and create a supragingival dataset
Note that I also ran the EMS statistics in two separate turnover scripts. The output from these will be loaded here.

```{r, admin, echo=FALSE, error=FALSE, warning=FALSE}
library("phyloseq");library("ggplot2");library(gridExtra);library("stringr");library("reshape2");library("genefilter"); library(knitr);library(DESeq2); theme_set(theme_bw());library(doBy);library(ggrepel)
```


## Discovery datatset
```{r}
discovery = readRDS("~/Dropbox/Periodontology2000/katie_biogeo_tree.RDS")
discovery

#how many subjects
levels(sample_data(discovery)$Subject)

#what's the median sequencing depth
summary(sample_sums(discovery))
```

## Description of data
```{r}
decontam.phy=readRDS("~/Desktop/Periodontol2000/periodontol_moresubjects.RDS")
decontam.phy = subset_samples(decontam.phy, Subject !="Control")
#how many samples are in the validation dataset?
nsamples(decontam.phy)

#how many subjects are in the validation dataset?
length(levels(sample_data(decontam.phy)$Subject))

#how many healthy aim 1 subjects are in the validation dataset?
aim1 = subset_samples(decontam.phy, Aim=="SA1" | Aim=="Pilot")
length(levels(sample_data(aim1)$Subject))

#how many healthy aim 3 subjects are in the validation dataset?
aim3 = subset_samples(decontam.phy, Aim=="SA3")
length(levels(sample_data(aim3)$Subject))

#how many subgingival samples
sub = subset_samples(decontam.phy, Habitat_Class=="Sub")
sub = prune_taxa(taxa_sums(sub) > 0, sub)
sub = prune_samples(sample_sums(sub) > 0, sub)

#what subgingival samples were collected
#what sample sites were sampled
sub = subset_samples(decontam.phy, Habitat_Class=="Sub")
levels(sample_data(sub)$Tooth_Number)

#how many supragingival samples
supra = subset_samples(decontam.phy, Habitat_Class=="Supra")


#how many ASVs are there
decontam.phy

#write a csv and contruct a highest taxonomic rank degination in the taxa table
df = data.frame(tax_table(decontam.phy))
#write.csv(df, file="periodontology2000_tax.csv")
newtax= read.csv("~/Desktop/Projects/Stanford/RelmanLab/Hyposalivation/Periodontol2000/periodontology2000_tax.csv", row.names = 1)
newtax = as.matrix(newtax)
newtax= tax_table(newtax)
tax_table(decontam.phy)= newtax


#replae taxa names with the highest label
df=data.frame(tax_table(decontam.phy))
taxa_names(decontam.phy)=df$Highest.Rank

#what is the median sequencing depth of the validation dataset?
summary(sample_sums(decontam.phy))


df = data.frame(sample_sums(decontam.phy), sample_data(decontam.phy))
colnames(df)[1] = "depth"

p = ggplot(df, aes(depth, color=Habitat_Class)) + geom_density()
p

mysum = doBy::summaryBy(depth~Habitat_Class, df, FUN=c(median, mean, length, sum, min, max))
mysum
```

### Description of data: Phyla level composition survey
- These results are described in the main text 
- Supplementary Table 1: Top 5 Phyla account for most of the reads

```{r}
#how many unique phyla are found?
length(get_taxa_unique(decontam.phy, taxonomic.rank="Phylum"))

#What is the relative abundance of each Phylum?
library(doBy)
phy <- tax_glom(decontam.phy, taxrank="Phylum")
      tax.count <- data.frame(taxa_sums(phy),tax_table(phy)[,2])
      rownames(tax.count)= NULL
      colnames(tax.count)[1] <- c("Abundance")
      tax.count$Percent <- round(tax.count$Abundance/sum(tax.count$Abundance)*100, 4)
      library(plyr)
      Phylum_df <- arrange(tax.count, desc(Percent))
      colnames(Phylum_df) = c("Total.Abundance", "Phylum", "Total.Percent")
      
      
      #how much do the top 5 phyla contribute to total abundance?
      top5 <- Phylum_df[1:5, ]
      round(sum(top5$Percent),2)

      #What are the top 5 Phyla?
      top5

##################what about the subgingival samples
subgingival.phy = subset_samples(decontam.phy, Habitat_Class=="Sub")
      phy <- tax_glom(subgingival.phy, taxrank="Phylum")
      tax.count <- data.frame(taxa_sums(phy),tax_table(phy)[,2])
      rownames(tax.count)= NULL
      colnames(tax.count)[1] <- c("Abundance")
      tax.count$Percent <- round(tax.count$Abundance/sum(tax.count$Abundance)*100, 4)
      Phylum_sub <- arrange(tax.count, desc(Percent))
      Phylum_sub <- subset(Phylum_sub, Abundance > 0)
      colnames(Phylum_sub) = c("Sub.Abundance", "Phylum" ,"Sub.Percent")
      
##################what about the supragingival samples
supragingival.phy = subset_samples(decontam.phy, Habitat_Class=="Supra")
      phy <- tax_glom(supragingival.phy, taxrank="Phylum")
      tax.count <- data.frame(taxa_sums(phy),tax_table(phy)[,2])
      rownames(tax.count)= NULL
      colnames(tax.count)[1] <- c("Abundance")
      tax.count$Percent <- round(tax.count$Abundance/sum(tax.count$Abundance)*100, 4)
      Phylum_supra <- arrange(tax.count, desc(Percent))
      Phylum_suora <- subset(Phylum_supra, Abundance > 0)
      colnames(Phylum_supra) = c("Supra.Abundance", "Phylum" ,"Supra.Percent")


#plot the phylum level differenes
df = join(Phylum_sub, Phylum_supra,  by="Phylum")
df= join(df, Phylum_df, by="Phylum")
dfm = melt(df, id.vars = c("Sub.Abundance", "Supra.Abundance", "Phylum", "Total.Abundance"))
fig1a = ggplot(dfm, aes(variable, value, fill=Phylum)) + geom_bar(position="stack", stat="identity") +
        scale_x_discrete(labels=c("Sub.Percent" = "Subgingival", "Supra.Percent" = "Supragingival", "Total.Percent" = "Totals")) +
        xlab("Habitat Class") + ylab("Percent Relative Abundance")

#test differences using proportion test
    testme = as.matrix(cbind(df[,3], df[,5]))
    prop.test(testme)
```




### Description of data: genus-level summary of data
- This is described in the main text
- Supplementary Table 2: Top 10 Genera account for most of the reads

```{r}
#What is the number of Genera found?
length(get_taxa_unique(decontam.phy, taxonomic.rank="Genus"))

#what are the abundance levels of each genus?
supra.genus <- tax_glom(decontam.phy, taxrank="Genus")
tax.count <- data.frame(tax_table(supra.genus)[,2:7], taxa_sums(supra.genus))
rownames(tax.count) = NULL
colnames(tax.count) <- c("Phylum","Class","Order","Family","Genus", "Species",  "Abundance")
tax.count$Percent <- round(tax.count$Abundance/sum(tax.count$Abundance)*100, 4)
library(plyr)
Genus_df <- arrange(tax.count, desc(Percent))
Genus_df <- Genus_df[order(Genus_df$Percent, decreasing=TRUE), ] 

#how much do the top 10 genera contribute to total abundance?
top10 <- Genus_df[1:10, ]
round(sum(top10$Percent),3)

#what about the top 5
top5 <- Genus_df[1:5, ]
round(sum(top5$Percent),3)

###How diverse are the top 10 genera? i.e., how many species are there per genus?
top10 <- as.vector(Genus_df$Genus[1:10])
Diversity.list <- vector("list", 10)
names(Diversity.list) <- top10

for(i in 1:length(top10)){
       physub = subset_taxa(decontam.phy, Genus==top10[i])
       physub = prune_taxa(taxa_sums(physub) > 0, physub)
       Diversity.list[[i]] <- physub
}

#compute the number of taxa in each element of the list
Ntaxa <- data.frame(unlist(lapply(Diversity.list, ntaxa)))

colnames(Ntaxa) <- "N.Species"
#Make a table with percent abundance and number of taxa
genus.tab <- data.frame(Genus_df[1:10,], Ntaxa)
library(knitr)
kable(genus.tab, format="markdown")
write.csv(genus.tab, file="TableS2.csv")

```


### how does alpha diversity vary?
```{r}
mi= subset_samples(decontam.phy, Tooth_Class %in% c("Incisor_Central", "Molar"))
mi = prune_samples(sample_sums(mi) > 0, mi)

subgingival.phy = subset_samples(mi, Habitat_Class=="Sub")
subgingival.phy = prune_taxa(taxa_sums(subgingival.phy) > 0, subgingival.phy)
subgingival.phy = prune_samples(sample_sums(subgingival.phy) > 0, subgingival.phy)


supragingival.phy = subset_samples(mi, Habitat_Class=="Supra")
supragingival.phy = prune_taxa(taxa_sums(supragingival.phy) > 0, supragingival.phy)
supragingival.phy = prune_samples(sample_sums(supragingival.phy) > 0, supragingival.phy)


meths = c("Shannon", "Chao1", "Simpson")
sites = levels(as.factor(sample_data(subgingival.phy)$Habitat))
sub.holder <-vector('list',length(sites)) 
supra.holder <- vector('list', length(sites))
prare.holder <- vector('list', length(sites))
brare.holder  <- vector('list', length(sites))

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(subgingival.phy, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(subgingival.phy))
  sub.holder[[i]] <- df
  
}

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(supragingival.phy, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(supragingival.phy))
  supra.holder[[i]] <- df
  
}

#rarefy the supragingival samples to 
set.seed(191)
supra.rare = rarefy_even_depth(supragingival.phy, sample.size=20000, replace=FALSE)

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(supra.rare, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(supra.rare))
  prare.holder[[i]] <- df
  
}

#rarefy the supragingival samples to 
set.seed(191)
sub.rare = rarefy_even_depth(subgingival.phy, sample.size=20000, replace=FALSE)
brare.holder <- vector('list', length(sites))

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(sub.rare, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(sub.rare))
  brare.holder[[i]] <- df
  
}
```


```{r}
df1 = do.call("rbind", sub.holder)
df2 = do.call("rbind", supra.holder)
df3 = do.call("rbind", prare.holder)
df3$Habitat_Class = "Rarefied_Supra"
df4 = do.call("rbind", brare.holder)
df4$Habitat_Class = "Rarefied_Sub"


df = data.frame(rbind(df1, df2, df3, df4))
dfm1 = melt(df, id.vars=colnames(sample_data(supragingival.phy)))
dfm1= subset(dfm1, variable !="se.chao1")
p1 = ggplot(dfm1, aes(Habitat_Class, value)) + facet_wrap(~variable, scales="free_y", ncol=4) + geom_boxplot()
p1

#test to see whether alpha diversity varies based on tooth class
df = subset(df, Habitat_Class %in% c("Rarefied_Supra", "Rarefied_Sub" ))
wilcox.test(Chao1~Habitat_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Shannon~Habitat_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Simpson~Habitat_Class, data=df, exact=TRUE, conf.int=TRUE)

```


## Figure 1, Difference between microbial habitat
- We also see in panel figure 1d, the taxa that differentiate the sub and supragingival habitats

```{r, Figure 1, fig.width=12, fig.height=9, warning=FALSE}
library(RColorBrewer)
### Create an index tooth biogeography subset
teeth = c("3", "8", "9", "14", "19", "24", "25", "30")
index_subsites <- subset_samples(decontam.phy, Tooth_Number %in% teeth)
index_subsites
        
#how many samples are in average 2 subjects? Looks like about 25% of samples
summary(sample_data(index_subsites)$Subject)

#filter the taxa to include those present in 25% of samples
filtergroup = filterfun(kOverA(k=70, A=1)) #k = number of samples; A = abundance
        index25 = filter_taxa(index_subsites, filtergroup, prune=TRUE) 
        index25 = prune_taxa(taxa_sums(index25) > 0, index25) 
        index25 = prune_samples(sample_sums(index25) > 0, index25) 
        index25
#convert all the data to relative abundance        
indexRA = transform_sample_counts(index25, function(x) x / sum(x))

#Do PCOA
index.pcoa <- ordinate(indexRA, method="PCoA", distance="bray")
evals <- index.pcoa$values$Eigenvalues

### Use DESEq2 to identify taxa that are differentially abundant between subgingival and supragingival sites
diagdds = phyloseq_to_deseq2(index25, ~ Habitat_Class)

# calculate geometric means prior to estimate size factors
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
diagdds = DESeq(diagdds, fitType="local")


res = results(diagdds, cooksCutoff = FALSE)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(index25)[rownames(sigtab), ], "matrix"))



posigtab = sigtab[sigtab[, "log2FoldChange"] > 5, ]
posigtab = posigtab[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")]

library("ggplot2")
sigtabgen = subset(sigtab, !is.na(Genus))
# Phylum order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))
# Genus order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))
sigtabgen = subset(sigtabgen, padj < 0.05)
fig1a=ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange, color=Phylum)) + 
      geom_vline(xintercept = 0.0, color = "gray", size = 0.5) +
        geom_point(size=6) + 
      theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

print(fig1a)
#look at all the data as a function of tooth class

fig1b=plot_ordination(indexRA, index.pcoa, type="samples", color="Habitat_Class") + 
        geom_point(size=2, alpha=0.1) +  facet_wrap(~Tooth_Aspect)+ coord_fixed(sqrt(evals[2] / evals[1]))+ 
        scale_color_discrete(name="Habitat Class",
                         breaks=c("Supra", "Sub"),
                         labels=c("Supragingival Plaque", "Subgingival Plaque"))
dat = fig1b$data
fig1c=plot_ordination(indexRA, index.pcoa, type="samples", color="Aim") + geom_point(size=2, alpha=0.1) + 
        theme_bw()  + facet_wrap(~Tooth_Aspect)+coord_fixed(sqrt(evals[2] / evals[1]))+ 
        scale_color_discrete(name="Health Condition",
                         breaks=c("SA1", "SA3"),
                         labels=c("Control", "Sjogren's"))

fig1d=plot_ordination(indexRA, index.pcoa, type="samples", color="Subject") + geom_point(size=2, alpha=0.1) + 
        theme_bw() + facet_wrap(~Tooth_Aspect)+coord_fixed(sqrt(evals[2] / evals[1]))+ 
        scale_color_discrete(name="Subject",
                         breaks=c("1-101", "1-102", "1-103", "1-104", "1-105", "1-106", "3-301", "3-302", "3-303"),
                         labels=c("Control 1", "Control 2", "Control 3", "Control 4", "Control 5", "Control 6",
                                  "Sjogren's 1", "Sjogren's 2", "Sjogren's 3"))

#plot the taxa
p = plot_ordination(indexRA, index.pcoa, "taxa")
    df = data.frame(p$data, tax_table(indexRA), taxa_sums(index25))
    colnames(df)[17] = "sum"
    rownames(df) = NULL
    df$Highest = paste0(df$Genus, " ", df$Species)
    df$Highest = str_replace_all(df$Highest, "NA", "")

    fig1e=ggplot(df, aes(Axis.1, Axis.2, label=Highest, color=Phylum))+coord_fixed(sqrt(evals[2] / evals[1]))+
          xlab("Axis 1 [24.1%]")  + ylab("Axis 2 [10.8%]") +  geom_text_repel(aes(Axis.1, Axis.2, label = Highest), size = 3)
    
  
grid.arrange(fig1b, fig1e, ncol=2)
ggsave(grid.arrange(fig1b, fig1e, ncol=1),  file="~/Desktop/Figure1.pdf",  width = 12, height = 10, units ="in",dpi = 300)
```

```{r}
fig1d=ggplot(dat, aes(Jaw_Quadrant, Axis.1), type="samples", color="Aim") + geom_point(size=2, alpha=0.1) + 
        theme_bw()  + facet_wrap(~Tooth_Class)+coord_fixed(sqrt(evals[2] / evals[1]))

```

Now let's quantify the segregation of samples using an adonis on Bray Curtis.

```{r}
library(vegan)
otus = data.frame(otu_table(index25))
otus.h = decostand(otus, "hell")
attach(sample_data(index25))
set.seed(12)
index.adonis <- adonis(otus.h ~Habitat_Class*Aim*Tooth_Class*Tooth_Aspect, permutations=1000, strata = Subject) 
df <- format(data.frame(index.adonis$aov.tab), digits=2)
df
detach(sample_data(index25))
```






## Figure 2: Do subgingival communities conform the gradient
Figure 2A: We see that communities differ by tooth class at supragingival sites, but not as much by subgingival sites. Interestingly, subgingival diversity appears to be most variability between molar sites within the healthy control group.

```{r, fig.width=6, fig.height=6, warning=FALSE} 
sub.n= c("14", "19", "24", "25", "3" , "30",  "8",  "9" )
#### 1. Compute the third-order orthogonal polynomial terms
index_subsites= subset_samples(decontam.phy,Tooth_Number %in% sub.n)
subs = levels(sample_data(index_subsites)$Habitat_Class)
otu.list <- vector("list", length(subs))
names(otu.list) <- subs

for(i in 1:length(subs)){


          #hellinger transform the data
          library(vegan)
          x1 = subset_samples(index_subsites, Habitat_Class==subs[[i]])
          x1 = prune_taxa(taxa_sums(x1) > 0, x1)
          x1 = prune_samples(sample_sums(x1) > 0, x1)
          otus = data.frame(otu_table(x1))
          otus.h = decostand(otus, "hellinger")
          map = sample_data(x1)
          map$x = as.numeric(as.character(map$x))
          map$y = as.numeric(as.character(map$y))
          
          #center the coordinates
          xygrid = cbind(map$x, map$y)
          xygrid.c <- scale(xygrid, scale=FALSE)
          
          ### Compute the third-order orthogonal polynomial function using centered geographic coordinates
          poly.xy3 <- poly(xygrid.c, degree = 3, raw=FALSE) #, coefs=TRUE) 
          colnames(poly.xy3) <- c("X", "X^2", "X^3", "Y", "XY", "X^2Y", "Y^2", "XY^2", "Y^3")
          poly.xy3.df <- data.frame(poly.xy3, map$x, map$y)
          
          library(ade4)
          #perform the RDA on the hellinger transformed data; extract the coordinates 
          rld.pca <- dudi.pca(otus.h , center=TRUE, scale=TRUE, scannf=FALSE, nf=10)
          rld.xy3 <- pcaiv(rld.pca, poly.xy3, scannf = FALSE, nf = 6)
          rld.xy3.df <- data.frame(rld.xy3$ls, map)
          xy3.df <- data.frame(rld.xy3.df, poly.xy3, sample_data(x1))
          
          # how much of the variance does the trend surface model explain?
          rld.xy3.var <- sum(rld.xy3$eig)/sum(rld.pca$eig)*100
          rld.xy3.var
          
          #look at the eigenvalues
          rld.xy3$eig
          
          #look at the screeplot
          screeplot(rld.xy3) #we should look at the first 3 axes
          
          #what is the percent of ewxplained variance
          Explainedvariance = rld.xy3$eig/sum(rld.xy3$eig)*100
          Explainedvariance
          
          #force x, y to numeric
          xy3.df$x <- as.numeric(as.character(xy3.df$x))
          xy3.df$y <- as.numeric(as.character(xy3.df$y))
          xy3.df$subsetsite = subs[[i]]
          otu.list[[i]] = xy3.df
          
}
xy3.df = do.call("rbind", otu.list)
### Force variables to order in ggplot
order=1:32
xy3.df$Tooth_Number <- factor(xy3.df$Tooth_Number, as.character(order))

#plot axis 1

xy3.df = subset(xy3.df, Aim=="SA1")
sub.df = subset(xy3.df, Habitat_Class=="Sub")
supra.df = subset(xy3.df, Habitat_Class=="Supra")


fig2a = ggplot(supra.df, aes(Tooth_Number, Axis1, group=Aim))  + theme_bw() + ylab("Axis 1") + 
      facet_wrap(Habitat_Class~Aim)+ xlab("Tooth")  + geom_point() + geom_smooth() + ggtitle("a)")+ 
      scale_x_discrete(breaks = seq(from=0, to=30, by=5), labels=seq(from=0, to=30, by=5))

fig2b = ggplot(sub.df, aes(Tooth_Number, Axis1, group=Aim))  + theme_bw() + ylab("Axis 1") + 
      facet_wrap(Habitat_Class~Aim)+ xlab("Tooth")  + geom_point() + geom_smooth() + ggtitle("b)")+ 
      scale_x_discrete(breaks = seq(from=0, to=30, by=5), labels=seq(from=0, to=30, by=5))

grid.arrange(fig2a, fig2b, ncol=2)

buccal = subset(sub.df, Tooth_Aspect=="Buccal")
lingual = subset(sub.df, Tooth_Aspect=="Lingual")
fig2c = ggplot(sub.df, aes(Tooth_Class, Axis1, fill=Tooth_Class))  + theme_bw() + ylab("Axis 1") + 
      facet_wrap(~Tooth_Aspect)+ xlab("Tooth")   + geom_violin() + ggtitle("b)")+ geom_point(aes(shape=Jaw_Quadrant))




ggsave(grid.arrange(fig2a, fig2b, ncol=2),  file="~/Desktop/Figure3.pdf",  width = 10, height = 6, units ="in",dpi = 300)
```


```{r}
#do we see a difference in sequencing depth by tooth location, no we don't for most subjects 

ling= subset(sub.df, Tooth_Aspect=="Lingual")
wilcox.test(Axis1~Tooth_Class, data=ling, exact=TRUE, conf.int=TRUE)

buc= subset(sub.df, Tooth_Aspect=="Buccal")
wilcox.test(Axis1~Tooth_Class, data=buc, exact=TRUE, conf.int=TRUE)
```

```{r}

```

## Figure 2: Do subgingival communities conform the gradient - discovery dataset
Figure 2A: We see that communities differ by tooth class at supragingival sites, but not as much by subgingival sites. Interestingly, subgingival diversity appears to be most variability between molar sites within the healthy control group.

```{r, fig.width=6, fig.height=6, warning=FALSE} 
discovery#subset on taxa present in 10% of samples
filtergroup = filterfun(kOverA(k=21, A=1)) #k = number of samples; A = abundance
        flows = filter_taxa(discovery, filtergroup, prune=TRUE) 
        flows = prune_taxa(taxa_sums(flows) > 0, flows)
        flows = prune_samples(sample_sums(flows) > 0, flows)
        flows

          #hellinger transform the data
          library(vegan)
          otus = data.frame(otu_table(flows))
          otus.h = decostand(otus, "hellinger")
          map = sample_data(flows)
          map$x = as.numeric(as.character(map$x))
          map$y = as.numeric(as.character(map$y))
          
          #center the coordinates
          xygrid = cbind(map$x, map$y)
          xygrid.c <- scale(xygrid, scale=FALSE)
          
          ### Compute the third-order orthogonal polynomial function using centered geographic coordinates
          poly.xy3 <- poly(xygrid.c, degree = 3, raw=FALSE) #, coefs=TRUE) 
          colnames(poly.xy3) <- c("X", "X^2", "X^3", "Y", "XY", "X^2Y", "Y^2", "XY^2", "Y^3")
          poly.xy3.df <- data.frame(poly.xy3, map$x, map$y)
          
          library(ade4)
          #perform the RDA on the hellinger transformed data; extract the coordinates 
          rld.pca <- dudi.pca(otus.h , center=TRUE, scale=TRUE, scannf=FALSE, nf=10)
          rld.xy3 <- pcaiv(rld.pca, poly.xy3, scannf = FALSE, nf = 6)
          rld.xy3.df <- data.frame(rld.xy3$ls, map)
          xy3.df <- data.frame(rld.xy3.df, poly.xy3)
          
          # how much of the variance does the trend surface model explain?
          rld.xy3.var <- sum(rld.xy3$eig)/sum(rld.pca$eig)*100
          rld.xy3.var
          
          #look at the eigenvalues
          rld.xy3$eig
          
          #look at the screeplot
          screeplot(rld.xy3) #we should look at the first 3 axes
          
          #what is the percent of ewxplained variance
          Explainedvariance = rld.xy3$eig/sum(rld.xy3$eig)*100
          Explainedvariance
          
          #force x, y to numeric
          xy3.df$x <- as.numeric(as.character(xy3.df$x))
          xy3.df$y <- as.numeric(as.character(xy3.df$y))
          xy3.df$subsetsite = subs[[i]]
          otu.list[[i]] = xy3.df
          
}
xy3.df = do.call("rbind", otu.list)
### Force variables to order in ggplot
order=1:32
xy3.df$Tooth_Number <- factor(xy3.df$Tooth_Number, as.character(order))

#plot axis 1
sub.df = subset(xy3.df, Habitat_Class=="Sub")
supra.df = subset(xy3.df, Habitat_Class=="Supra")

fig2a = ggplot(sub.df, aes(Tooth_Number, Axis1, group=Aim))  + theme_bw() + ylab("Axis 1") + 
      facet_wrap(Habitat_Class~Aim)+ xlab("Tooth")  + geom_point(aes(color=Tooth_Class))+ geom_smooth() + ggtitle("a)")+ 
      scale_x_discrete(breaks = seq(from=0, to=30, by=2), labels=seq(from=0, to=30, by=2)) + geom_vline(aes(xintercept=15))


fig2b = ggplot(supra.df, aes(Tooth_Number, Axis1, group=Aim))  + theme_bw() + ylab("Axis 1") + 
      facet_wrap(Habitat_Class~Aim)+ xlab("Tooth")  + geom_point(aes(color=Tooth_Class)) + geom_smooth() + ggtitle("b)")+ 
      scale_x_discrete(breaks = seq(from=0, to=30, by=5), labels=seq(from=0, to=30, by=5))



grid.arrange(fig2a, fig2b, ncol=2)
ggsave(grid.arrange(fig2a, fig2b, ncol=2),  file="~/Desktop/Figure3.pdf",  width = 10, height = 6, units ="in",dpi = 300)
```



## Figure 4: Covariation of sub and supragingival communities


```{r}
#let's clean the data up so that we retain only samples present in an individual mouth at both sub/supra sites
subgingival.phy = subset_samples(decontam.phy, Habitat_Class=="Sub")
foo = sample_data(subgingival.phy)$Habitat
foo = colsplit(foo, "([_])", c("junk", "Habitat"))
sample_data(subgingival.phy)$Habitat.specific = foo$Habitat

#subgingival
    P1.1a = subset_samples(subgingival.phy, Subject=="1-101") #20
    P1.2a = subset_samples(subgingival.phy, Subject=="1-102") #22
    P1.3a = subset_samples(subgingival.phy, Subject=="1-104") #24
    P1.4a = subset_samples(subgingival.phy, Subject=="1-105") #24
    P1.5a =  subset_samples(subgingival.phy, Subject=="1-106" & Replicate=="R00") #19
    P1.5a = subset_samples(P1.5a, Habitat !="Sub_09L")
    P1.6a = subset_samples(subgingival.phy, Subject=="3-301") #24
    P1.7a = subset_samples(subgingival.phy, Subject=="3-302") #24
    P1.8a = subset_samples(subgingival.phy, Subject=="3-303") #
    statis_subgingival = merge_phyloseq(P1.1a, P1.2a, P1.3a, P1.4a, P1.5a, P1.6a, P1.7a, P1.8a)
    statis_subgingival= prune_samples(sample_sums(statis_subgingival) > 0, statis_subgingival)
    statis_subgingival
    
keep = levels(sample_data(subgingival.phy)$Tooth_Number)
supragingival.phy = subset_samples(decontam.phy, Habitat_Class=="Supra")
supragingival.phy=subset_samples(supragingival.phy, Tooth_Number %in%  keep)
foo = sample_data(supragingival.phy)$Habitat
foo = colsplit(foo, "([_])", c("junk", "Habitat"))
sample_data(supragingival.phy)$Habitat.specific = foo$Habitat


#supragingival
    P1.1b = subset_samples(supragingival.phy, Subject=="1-101"  &  !(Habitat %in% c("Supra_08B", "Supra_08L" ,"Supra_09B" ,"Supra_09L" )))
    P1.2b = subset_samples(supragingival.phy, Subject=="1-102" & !(Habitat %in% c("Supra_27L"))) #25
    P1.3b = subset_samples(supragingival.phy, Subject=="1-104") #24
    P1.4b = subset_samples(supragingival.phy, Subject=="1-105") #24
    P1.5b =  subset_samples(supragingival.phy, Subject=="1-106"  & !(Habitat %in% c( "Supra_27B", "Supra_27L", "Supra_30B" ,"Supra_30L")))
    P1.6b = subset_samples(supragingival.phy, Subject=="3-301") #24
    P1.7b = subset_samples(supragingival.phy, Subject=="3-302") #24
    P1.8b = subset_samples(supragingival.phy, Subject=="3-303") #& !(Habitat %in% c("Supra_06B" "Supra_06L"
    
    
    statis_supragingival = merge_phyloseq(P1.1b, P1.2b, P1.3b, P1.4b, P1.5b, P1.6b, P1.7b, P1.8b)
    statis_supragingival = subset_samples(statis_supragingival, Replicate !="R04")
    statis_supragingival= prune_samples(sample_sums(statis_supragingival) > 0, statis_supragingival)

#make a new phyloseq object     
statis = merge_phyloseq(statis_supragingival, statis_subgingival)
#remove taxa present in fewert han 25% of samples    
filtergroup = filterfun(kOverA(k=70, A=10)) #k = number of samples; A = abundance
        sub.filt = filter_taxa(statis, filtergroup, prune=TRUE) 
        sub.filt = prune_samples(sample_sums(sub.filt) > 0, sub.filt) 
        sub.filt

```

```{r}
# read in the perio data
s101p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Bopcal_SA1/CSV/REDCap17_ESE_bopcal_SA1_v5.2_20140731_DD.csv")
s102p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Bopcal_SA1/CSV/REDCap18_ESE_bopcal_SA1_v5.2_20140805_DD.csv")
s104p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Bopcal_SA1/CSV/REDCap39_ESE_bopcal_SA1_v5.2_20140925_DD.csv")
s105p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Bopcal_SA1/CSV/REDCap49_ESE_bopcal_SA1_v5.2_20141021_DD.csv")
s106p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Bopcal_SA1/CSV/REDCap63_ESE_bopcal_SA1_v5.2_20141104_DD.csv")
s301p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Bopcal_SA3/CSV/REDCap66_ESE_bopcal_SA3_v5.2_20141120_DD.csv")
s302p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Bopcal_SA3/CSV/REDCap178_ESE_bopcal_SA3_v5.2_20150528_DD.csv")
s303p = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Bopcal_SA3/CSV/REDCap219_ESE_bopcal_SA3_v5.2_20150818_SK.csv")

perio.df = data.frame(rbind(s101p, s102p, s104p, s105p, s106p, s301p, s302p, s303p))
perio.df = subset(perio.df, pd !="Missing")
perio.df = subset(perio.df, tooth !="NA")

pd = doBy::summaryBy(pd~redcap+tooth, data=perio.df, FUN=mean)
colnames(pd) = c("redcap", "Tooth_Number", "PD")

cal = doBy::summaryBy(cal~redcap+tooth, data=perio.df, FUN=mean)
colnames(cal) = c("redcap", "Tooth_Number", "CAL")

bop = doBy::summaryBy(percent.bop~redcap+tooth, data=perio.df, FUN=mean)
colnames(bop) = c("redcap", "Tooth_Number", "BOP")

#read in the caries data #
s101c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Axium_Data_SA1/CSV/REDCap17_ESE_Axium_SA1_v1.2_20140731_DD.csv")
s102c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Axium_Data_SA1/CSV/REDCap18_ESE_Axium_SA1_v1.2_20140819_DP.csv")
s104c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Axium_Data_SA1/CSV/REDCap39_ESE_Axium_SA1_v1.2_20140925_DD.csv")
s105c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Axium_Data_SA1/CSV/REDCap49_ESE_Axium_SA1_v1.2_20141021_DD.csv")
s106c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA1_data/Axium_Data_SA1/CSV/REDCap63_ESE_Axium_SA1_v1.2_20141104_DD.csv")
s301c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Axium_Data_SA3/CSV/REDCap66_ESE_Axium_SA3_v1.2_20141120_DD.csv")
s302c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Axium_Data_SA3/CSV/REDCap178_ESE_Axium_SA3_v1.2_20150528_DD.csv")
s303c = read.csv("~/Dropbox/Hyposalivation_UCSF-StanfordFileSharing/SA3_data/Axium_Data_SA3/CSV/REDCap219_ESE_Axium_SA3_v1.2_20150818_SK.csv")

caries.df = data.frame(rbind(s101c, s102c, s104c, s105c, s106c, s301c, s302c, s303c))
caries.df = subset(caries.df, description !="MISSING")
caries.df = subset(caries.df, tooth !="NA")


caries.df$binary = ifelse(caries.df$description =="SOUND", 0, 1)
dmfs = summaryBy(binary~redcap + tooth, data=caries.df, FUN=sum)
colnames(dmfs) =c("redcap", "Tooth_Number",  "dmfs")


map = data.frame(sample_data(sub.filt))
pids = read.csv("~/Desktop/pids.csv")
colnames(pids) = c("redcap", "Subject")
flow = read.csv("~/Dropbox/Periodontology2000/flowrates.csv")

new.map = join(map, pids, "Subject")
new.map = join(new.map, pd)
new.map = join(new.map, cal)
new.map = join(new.map, bop)
new.map = join(new.map, flow)
new.map = join(new.map, dmfs)
rownames(new.map) = new.map$X.SampleID

sample_data(sub.filt) = sample_data(new.map)

```

### Look at the clinical data
```{r, fig.height=8, fig.width=12} 
#set an analysis up for CCA
otus = otu_table(sub.filt)
otus = decostand(otus, "hell")
map1 = data.frame(sample_data(sub.filt)$Habitat_Class, sample_data(sub.filt)$UWS_FR, sample_data(sub.filt)$CAL, 
                  sample_data(sub.filt)$PD, sample_data(sub.filt)$BOP, sample_data(sub.filt)$dmfs)
                  
colnames(map1) = c("Class", "uws_fr",  "cal", "pd", "bop", "dmfs")
map1$Class = as.numeric(map1$Class)
subs = as.vector(sample_data(sub.filt)$Subject)
runs = as.vector(sample_data(sub.filt)$Pool_Name)

#for some reason this wworks on the command line but not in knitr
attach(map1)
flow.cca = vegan::cca(otus ~ map1$Class* map1$uws_fr* map1$cal* map1$pd* map1$bop* map1$dmfs, sitenv=map1)


#run an anova on this
set.seed(100)
flow.aov = anova.cca(flow.cca, by="term", strata=subs)
flow.aov

#how much variance does the model explain
eig = flow.cca$CCA$eig
eigp = 100*(eig/sum(eig))
head(eigp)


#get the sample site scores and plot them
sites = data.frame(flow.cca$CCA$wa, sample_data(sub.filt))
sites$Aim = ifelse(sites$Aim=="SA1", "Control", "Sjogrens")


#convert the MFS score to factor
p1 = ggplot(sites, aes(CCA1, CCA2, color=as.factor(Aim))) + geom_point() + guides(color=guide_legend(title="Health Status")) + 
            ggtitle("a)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1])) 

p2 = ggplot(sites, aes(CCA1, CCA2, color=uws_fr)) + geom_point() + guides(color=guide_legend(title="uws_fr")) + 
            ggtitle("b)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1]))+ scale_colour_gradientn(colours=c("red", "blue"))

p3 = ggplot(sites, aes(CCA1, CCA2, color=dmfs)) + geom_point() + guides(color=guide_legend(title="dmfs")) + 
            ggtitle("c)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1]))+ scale_colour_gradientn(colours=c("red", "blue"))

p4 = ggplot(sites, aes(CCA1, CCA2, color=CAL)) + geom_point() + guides(color=guide_legend(title="CAL")) + 
            ggtitle("d)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1]))+ scale_colour_gradientn(colours=c("red", "blue"))

p5 = ggplot(sites, aes(CCA1, CCA2, color=BOP)) + geom_point() + guides(color=guide_legend(title="BOP")) + 
            ggtitle("e)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1]))+ scale_colour_gradientn(colours=c("red", "blue"))

p6 = ggplot(sites, aes(CCA1, CCA2, color=PD)) + geom_point() + guides(color=guide_legend(title="PD")) + 
            ggtitle("f)")  + xlab("CCA1 (27.7%)") + ylab("CCA2 (19.3%)") + 
            theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +
            coord_fixed(sqrt(eigp[2] / eigp[1]))+ scale_colour_gradientn(colours=c("red", "blue"))

grid.arrange(p1, p2, p3, p4, p5, p6, ncol=2)
```


#### Look at the covariation between subgingival and supragingival sites and taxa

```{r, fig.height=10, fig.width=12}
#subset on supragingival
supra = subset_samples(sub.filt, Habitat_Class=="Supra")
supra = prune_taxa(taxa_sums(supra) >1, supra)

#subset on subgingival
sub = subset_samples(sub.filt, Habitat_Class=="Sub")
sub = prune_taxa(taxa_sums(sub) >1, sub)


#transform
  Hstatis_subgingival = decostand(data.frame(otu_table(sub)), "hel")
  Hstatis_supragingival = decostand(data.frame(otu_table(supra)), "hel")

#generate the model
set.seed(981)
subs = sample_data(supra)$Subject
mod1 = CCorA(Hstatis_subgingival, Hstatis_supragingival, permutations=999)
biplot(mod1, plot.type="ov")

  eval = 100*(mod1$Eigenvalues/sum(mod1$Eigenvalues))


#plot subgingival samples
df1 = data.frame(mod1$Cx, sample_data(sub))
p1=ggplot(df1, aes(CanAxis1, CanAxis2, label=Subject, color=Aim)) + geom_text() + ggtitle(("a)")) + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
          theme(legend.position="none")

#plot supragingival samples
df2 = data.frame(mod1$Cy, sample_data(supra))
p2 = ggplot(df2, aes(CanAxis1, CanAxis2, label=Subject, color=Aim)) + geom_text() + ggtitle("c)") + 
         theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
        theme(legend.position="none")


#subgingival species
df3 = data.frame(mod1$corr.Y.Cy, tax_table(sub))
#write.csv()
myTax3 = df3$Highest.Rank
p3 = ggplot(df3, aes(CanAxis1, CanAxis2, label=rownames(df3), color=Phylum))  + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1))  + 
          theme(legend.position="none") + geom_text_repel(aes(CanAxis1, CanAxis2, label = myTax3), size = 3) + ggtitle("b)")



#supragingival species
df4 = data.frame(mod1$corr.X.Cy, tax_table(supra))
myTax4 = df4$Highest.Rank
p4 = ggplot(df4, aes(CanAxis1, CanAxis2, label=rownames(df4), color=Phylum))  + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
          theme(legend.position="none")+ geom_text_repel(aes(CanAxis1, CanAxis2, label = myTax4), size = 3)+ ggtitle("d)")


#plot
grid.arrange(p1, p2, p3, p4, ncol=2)
ggsave(grid.arrange(p1, p2, p3, p4, ncol=2), file="~/Desktop/Figure2.png",  width = 12, height = 8.5, units ="in",dpi = 300, device="png")  
mod1

#look at s.mutans
sm = subset_taxa(sub.filt, Highest.Rank=="Streptococcus_mutans")

df= data.frame(otu_table(sm), sample_data(sm))
dfm= melt(df, id.vars = colnames(sample_data(sm)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)


p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)
p

#Prevotella_histicola
ph = subset_taxa(sub.filt, Highest.Rank=="Prevotella_histicola")

df= data.frame(otu_table(ph), sample_data(ph))
dfm= melt(df, id.vars = colnames(sample_data(ph)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)


p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)
p


#look at Scardovia_wiggsiae
sm = subset_taxa(sub.filt, Highest.Rank=="Scardovia_wiggsiae")

df= data.frame(otu_table(sm), sample_data(sm))
dfm= melt(df, id.vars = colnames(sample_data(sm)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)


p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)
p


#look at Streptococcus_sanguinis
sm = subset_taxa(sub.filt, Highest.Rank=="Streptococcus_sanguinis")

df= data.frame(otu_table(sm), sample_data(sm))
dfm= melt(df, id.vars = colnames(sample_data(sm)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)


p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)
p

```

### drop SS samples
```{r}
#subset on supragingival
supra = subset_samples(sub.filt, Habitat_Class=="Supra")
supra = subset_samples(supra, Aim =="SA1")
supra = prune_taxa(taxa_sums(supra) >1, supra)

#subset on subgingival
sub = subset_samples(sub.filt, Habitat_Class=="Sub")
sub = subset_samples(sub, Aim=="SA1")
sub = prune_taxa(taxa_sums(sub) >1, sub)

#transform
  Hstatis_subgingival = decostand(data.frame(otu_table(sub)), "hel")
  Hstatis_supragingival = decostand(data.frame(otu_table(supra)), "hel")

  
  eval = 100*(mod1$Eigenvalues/sum(mod1$Eigenvalues))

#generate the model
set.seed(981)
subs = sample_data(supra)$Subject
mod1 = CCorA(Hstatis_subgingival, Hstatis_supragingival, permutations=999)
biplot(mod1, plot.type="ov")


#plot subgingival samples
df1 = data.frame(mod1$Cx, sample_data(sub))
p1=ggplot(df1, aes(CanAxis1, CanAxis2, label=Subject, color=Subject)) + geom_text() + ggtitle(("a)")) + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
          theme(legend.position="none") + xlab("3.1%")+ ylab("3.1%")

#plot supragingival samples
df2 = data.frame(mod1$Cy, sample_data(supra))
p2 = ggplot(df2, aes(CanAxis1, CanAxis2, label=Subject, color=Subject)) + geom_text() + ggtitle("b)") + 
         theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
        theme(legend.position="none") + xlab("3.1%")+ ylab("3.1%")


#subgingival species
df3 = data.frame(mod1$corr.Y.Cy, tax_table(sub))
#write.csv()
myTax3 = df3$Highest.Rank
p3 = ggplot(df3, aes(CanAxis1, CanAxis2, label=rownames(df3), color=Phylum))  + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1))  + 
          theme(legend.position="bottom") + geom_text_repel(aes(CanAxis1, CanAxis2, label = myTax3), size = 3)



#supragingival species
df4 = data.frame(mod1$corr.X.Cy, tax_table(supra))
myTax4 = df4$Highest.Rank
p4 = ggplot(df4, aes(CanAxis1, CanAxis2, label=rownames(df4), color=Phylum))  + 
          theme_bw()+theme(text = element_text(size=20), axis.text.x = element_text(angle=0, vjust=1)) + 
          theme(legend.position="bottom")+ geom_text_repel(aes(CanAxis1, CanAxis2, label = myTax4), size = 3)


#plot
grid.arrange(p1, p2, p3, p4, ncol=2)
mod1

#look at s.mutans
sm = subset_taxa(sub.filt, Highest.Rank=="Streptococcus_mutans")

df= data.frame(otu_table(sm), sample_data(sm))
dfm= melt(df, id.vars = colnames(sample_data(sm)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)


p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)


#
ph = subset_taxa(sub.filt, Highest.Rank=="Prevotella_histicola")

df= data.frame(otu_table(ph), sample_data(ph))
dfm= melt(df, id.vars = colnames(sample_data(ph)))
ordering= c("3", "6","8","9", "11", "14","19", "22","24","25","27","30")
dfm$Tooth_Number  <- factor(dfm$Tooth_Number, levels = ordering)

p=ggplot(dfm,aes(Habitat_Class,value)) + geom_bar(stat="identity") + facet_wrap(~Aim)

p
p = ggplot(dfm,aes(Tooth_Number,PD, size=value, color=Tooth_Class)) + geom_point() + facet_wrap(Habitat_Class~Subject, ncol=8)

p
```


